spec/unit/spark_api/request_spec.rb

```diff
@@ -383,18 +392,17 @@ def reauthenticated
         r
       end
       it "should fail horribly on a get" do
-        expect { subject.get('/system')}.to raise_error(SparkApi::PermissionDenied) do |e| 
+        expect { subject.get('/system')}.to raise_error(SparkApi::PermissionDenied) { |e|
```

Comment:
The do/end here isn't right for some reason.  Changed to  braces and the block actually gets validated.  
Once that's fixed, the e.request_path validation was failing:

expected: "/system"
got: #<URI::HTTP http:/v1/system?ApiSig=SignedToken&AuthToken=EXPIRED>

Reply:
Why would it be a problem here, but not here? https://github.com/sparkapi/spark_api/pull/169/files#diff-09f333ddb2ed0043e81aa1bdd06ccffb985835cfed7f35916ce0707615b5b39cL395-L398

Very strange, but I wouldn't explore beyond ensuring that it's not doing something odd. Perhaps change the expectation to ensure it indeed fails like you'd want on a mismatch.

Reply:
Hmm not sure what to make of that.  Now when I'm testing things, the block is being executed either way.  Validated on both ruby 2.5 and ruby 3.0.  I guess the real question is what should the request_path be?  It's an object, not a string.  Looking at how it gets set I'm not sure how these tests were passing before if they're actually executed.

Reply:
This must be something that changed on one of the other gem upgrades.  On master, if I run this test 
      it "should fail horribly on a get" do
        expect { subject.get('/system')}.to raise_error(SparkApi::PermissionDenied) { |e|
          e.code.should == SparkApi::ResponseCodes::SESSION_TOKEN_EXPIRED
          e.request_path.should == '/systemNOPE'
        }
        subject.reauthenticated.should == 2

      end
     
     it passes with the do/end and fails (as it should) with { }

Reply:
&tldr; This test has always been invalid.  With the gem upgrades it turns out the do/end vs braces doens't matter.  The change from `== '/system'` to `.includes('system')` is sound.